src/components/product-progress/handlers/extract-pnp-progress.handler.ts

```diff
@@ -125,25 +126,36 @@ export class ExtractPnpProgressHandler {
             if (!(error instanceof StepNotPlannedException)) {
               continue;
             }
-            const stepLabel = ProductStep.entry(error.step).label;
             // kinda. close enough, I think, we give the cell ref as well.
             const goalLabel = extracted.bookName ?? extracted.story;
-            result.addProblem({
-              severity: 'Error',
-              groups: [
-                'Step is not planned',
-                `_${goalLabel}_ has progress reported on steps that have not been declared to be worked in this engagement`,
-                `_${goalLabel}_ has not declared _${stepLabel}_ \`${extracted.cell.ref}\` as a step that will be worked in this engagement`,
-              ],
-              message: oneLine`
-                Please update the goal in CORD to mark this step as planned
-                or upload an updated PnP file to the "Planning Spreadsheet" on the engagement page.
-              `,
-              source: extracted.cell,
+            result.addProblem(StepNotPlanned, extracted.cell, {
+              goal: goalLabel,
+              step: error.step,
             });
           }
         }
       }),
     );
   }
 }
+
+const StepNotPlanned = PnpProblemType.register({
```

Comment:
Why does StepNotPlanned fall under a handler and not in it's own file like the other errors in the pnp folder?